### DISCUSSION TITLE
How copy text from specific file into template generated file?

### DISCUSSION THREAD

**Discussion by executive-of-dysfxn** [SilentVoid13/Templater#733] [Help]
I'm very new to Templater and just starting to watch tutorials on javascript, so this might be a basic question and I just don't know the vocabulary to search for an answer. Is it possible to have a template where I'm prompted to select an option that will pull in text from another markdown file or other file type?

Let's say I am writing a summary of a discussion and it usually ends with one of a few outcomes. Each outcome has it's own paragraph of text as a note in my vault. I could have a list to transclude text from those other notes. Like `<% tp.system.suggester(["outcome1", "outcome2"], ["![[outcome1-text]]", "![[outcome2-text]]"]) %>`. I've tested this and I think it'll work for some templates I want to make.

The problem is if I want to edit any of that pre-written text in the new file. It would make more sense to me to copy text from `outcome1-text` when generating a new note from a template and edit in the new file as needed. What I don't know is how to make that happen. Would the text I want to copy need to be in a markdown file? .js file? This is all over my head so hopefully the question makes sense!

**Comment by AB1908** [comment]
Excellent idea. You'll want to use the Obsidian API to read contents of the file and push it into the current note. `app.vault.read("path to file.md")` gets you the contents of the file.

**Reply by executive-of-dysfxn** [reply]
It's taken me a while to come back to this, thanks for the tip! I haven't looked into the Obsidian API before, I'll have to see how that works. It occurred to me that I could just have another template and drop it in as needed to have text I can edit. At least until I figure out how to do something more sophisticated 😊
